devicetypes/smartthings/inovelli-dimmer.src/inovelli-dimmer.groovy

```diff
@@ -0,0 +1,599 @@
+			state "default", label:'${currentValue} W'
+		}
+		valueTile("energy", "device.energy", width: 2, height: 2) {
+			state "default", label:'${currentValue} kWh'
+		}
+		standardTile("refresh", "device.switch", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label:"", action:"refresh.refresh", icon:"st.secondary.refresh"
+		}
+	}
+
+	main(["switch","power","energy"])
+	details(["switch", "power", "energy", "refresh"])
+
+	preferences {
+		// Preferences template begin
+		parameterMap.each {
+			input (
+				title: it.name,
+				description: it.description,
+				type: "paragraph",
+				element: "paragraph"
+			)
+
+			switch(it.type) {
+				case "boolRange":
+					input(
+						name: it.key + "Boolean",
+						type: "bool",
+						title: "Enable",
+						description: "If you disable this option, it will overwrite setting below.",
+						defaultValue: it.defaultValue != it.disableValue,
+						required: false
+					)
+					input(
+						name: it.key,
+						type: "number",
+						title: "Set value (range ${it.range})",
+						defaultValue: it.defaultValue,
+						range: it.range,
+						required: false
+					)
+					break
+				case "boolean":
+					input(
+						type: "paragraph",
+						element: "paragraph",
+						description: "Option enabled: ${it.activeDescription}\n" +
+							"Option disabled: ${it.inactiveDescription}"
+					)
+					input(
+						name: it.key,
+						type: "bool",
+						title: "Enable",
+						defaultValue: it.defaultValue == it.activeOption,
+						required: false
+					)
+					break
+				case "enum":
+					input(
+						name: it.key,
+						title: "Select",
+						type: "enum",
+						options: it.values,
+						defaultValue: it.defaultValue,
+						required: false
+					)
+					break
+				case "range":
+					input(
+						name: it.key,
+						type: "number",
+						title: "Set value (range ${it.range})",
+						defaultValue: it.defaultValue,
+						range: it.range,
+						required: false
+					)
+					break
+			}
+		}
+		// Preferences template end
+	}
+}
+
+def installed() {
+	// Device-Watch simply pings if no device events received for 32min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+
+	// Preferences template begin
+	state.currentPreferencesState = [:]
+	parameterMap.each {
+		state.currentPreferencesState."$it.key" = [:]
+		state.currentPreferencesState."$it.key".value = getPreferenceValue(it)
+		if (it.type == "boolRange" && getPreferenceValue(it) == it.disableValue) {
+			state.currentPreferencesState."$it.key".status = "disablePending"
+		} else {
+			state.currentPreferencesState."$it.key".status = "synced"
+		}
+	}
+	readConfigurationFromTheDevice();
+// Preferences template end
+	createChildDevice("smartthings", "Child Color Selection", "${device.deviceNetworkId}:" + 3, "LEDColorConfiguration", "LEDColorConfiguration")
```

Comment:
We usually start component names with a lowercase letter so I'd suggest `ledBar` to match the user guide. Then the component label is intended to be more user friendly so I'd suggest `LED Bar`

Reply:
Instead of ` "${device.deviceNetworkId}:" + 3` I'd use ` "${device.deviceNetworkId}:3"`

Reply:
Also, do you create second (or first, if we stick to dni assigning convention) child device anywhere?

Reply:
Done